vpc/setup/setup_linux.go

```diff
@@ -144,15 +144,15 @@ func configureQdiscsForLink(ctx *context.VPCContext, link netlink.Link) error {
 }
 
 func setupIFBs(ctx *context.VPCContext) error {
-	err := setupIFB(ctx, vpc.IngressIFB, "classifier_ingress")
+	err := setupIFB(ctx, vpc.IngressIFB, "classifier_ingress", unix.ETH_P_IP)
 	if err != nil {
 		return err
 	}
 
-	return setupIFB(ctx, vpc.EgressIFB, "classifier_egress")
+	return setupIFB(ctx, vpc.EgressIFB, "classifier_egress", unix.ETH_P_ALL)
```

Comment:
it seems that for `!unix.ETH_P_ALL`, the bpf filter is a noop and will always do `TC_ACT_OK`, or are you relying on it to validate the L2 and L3 header lengths and formats?

Reply:
That should never happen (we should never get non-ethernet traffic on the interface veth that goes into the ifb).

Reply:
my bad, I meant `!unix.ETH_P_IP`

Reply:
Yes, except for the block rule, it passes through all IP traffic. The amount of non-IP traffic we receive should be minimal, but it is on my long-term TODO list (there's a JIRA for it), to make this capture and classify all container traffic, not just IPv4 traffic.